#
# Partial fix for: https://bugs.gnupg.org/gnupg/issue2144.
#
# The patch comes from the upstream. Fix detecting Solaris operating system.
# http://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=commitdiff;h=f7a77c5c236ecec846de9be46703026f9b01008f
#
--- a/src/gen-posix-lock-obj.c
+++ b/src/gen-posix-lock-obj.c
@@ -43,7 +43,8 @@
 #endif
 
 /* Special requirements for certain platforms.  */
-#if defined(__solaris__) && !defined (__LP64__) && !defined(_LP64)
+#if defined(__sun) && !defined (__LP64__) && !defined(_LP64)
+/* Solaris on 32-bit architecture.  */
 # define USE_DOUBLE_FOR_ALIGNMENT 1
 #else
 # define USE_DOUBLE_FOR_ALIGNMENT 0
